commit 9251b4403e85dca2ee4b6afee307498eaab52507
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-09-25T15:04:45-07:00 (12 days ago)
    
    joyent/sdc-net-agent#3 Instance loses NIC after sdc-migrate - again

diff --git a/lib/vmadm-watcher-fsm.js b/lib/vmadm-watcher-fsm.js
index 02ae759..e67555b 100644
--- a/lib/vmadm-watcher-fsm.js
+++ b/lib/vmadm-watcher-fsm.js
@@ -51,7 +51,6 @@ function VmadmEventsFSM(opts) {
         component: 'vmadm-events'
     }, true);
     self.vms = {};
-    self.ignore = {};
     self.emitter = null;
     self.stopWatcher = null;
     self.vmadm = opts.vmadm;
@@ -169,11 +168,9 @@ VmadmEventsFSM.prototype.handleEvent = function (ev) {
     self.log.trace({ev: ev}, 'saw event from "vmadm events"');
 
     if (ev.vm && ev.vm.do_not_inventory) {
-        self.ignore[ev.zonename] = true;
         self.log.debug('VM %s ignored - do_not_inventory set',
             ev.zonename);
-    } else if (ev.vm && !ev.vm.do_not_inventory) {
-        delete self.ignore[ev.zonename];
+        return;
     }
 
     switch (ev.type) {
@@ -192,10 +189,6 @@ VmadmEventsFSM.prototype.handleEvent = function (ev) {
 
         self.vms[ev.zonename] = ev.vm;
 
-        if (self.ignore[ev.zonename]) {
-            break;
-        }
-
         var changes = ev.changes.filter(function (change) {
             return (WATCHED_FIELDS.indexOf(change.path[0]) >= 0);
         });
@@ -213,12 +206,9 @@ VmadmEventsFSM.prototype.handleEvent = function (ev) {
         break;
     case 'delete':
         delete self.vms[ev.zonename];
-        if (!self.ignore[ev.zonename]) {
-            self.log.debug('VM %s deleted - setting needsUpdate',
-                ev.zonename);
-            needsUpdate = true;
-        }
-        delete self.ignore[ev.zonename];
+        self.log.debug('VM %s deleted - setting needsUpdate',
+            ev.zonename);
+        needsUpdate = true;
         break;
     default:
         assert.fail('unknown vmadm event type: ' + ev.type);
diff --git a/package.json b/package.json
index 95362cf..99b168a 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "net-agent",
     "description": "Triton Network Agent",
-    "version": "2.3.1",
+    "version": "2.3.2",
     "author": "Joyent (joyent.com)",
     "private": true,
     "dependencies": {
